site/src/components/roadmap/Roadmap.tsx

```diff
@@ -0,0 +1,220 @@
+    }
+  });
+  return sortedData;
+}
+
+function RoadmapCard(props: CardProps) {
+  return (
+    <Card
+      className={styles.spacing}
+      style={{ width: "250px", marginBottom: "15px", minHeight: "unset" }}
+      {...props}
+    />
+  );
+}
+
+export const Roadmap = ({ title, children }: RoadmapProps) => {
+  const [jsonData, setJsonData] = useState("");
+  const [roadmapData, setRoadmapData] = useState<any[]>([]);
+  const sortedRoadmapData = sortRoadmapDataByDate(roadmapData);
+  const [searchQuery, setSearchQuery] = useState("");
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        //gets the data
+        const response = await fetch("/api/roadmap");
+        const responseData = await response.json();
+        const jsonString = JSON.stringify(responseData, null, 2);
+        setJsonData(jsonString);
+
+        const items =
+          responseData?.data?.organization?.repository?.projectV2?.items?.nodes;
+
+        //creates an array of objects with data from github
+        const extractedData: RoadmapDataItem[] = items?.map((item: any) => {
+          const fieldValueNodes = item?.fieldValues?.nodes;
+          const text = getFieldValueByName(fieldValueNodes, "Title");
+          const startDate = getFieldValueByName(fieldValueNodes, "Start Date");
+          const targetDate = getFieldValueByName(
+            fieldValueNodes,
+            "Target Date"
+          );
+          const status = getFieldValueByName(fieldValueNodes, "Status");
+          const issueUrl = item?.content?.url;
+
+          return { text, startDate, targetDate, status, issueUrl };
+        });
+
+        //sets the data to retrieved data or null
+        setRoadmapData(extractedData || []);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const getFieldValueByName = (fieldValueNodes: any[], fieldName: string) => {
+    const fieldValueNode = fieldValueNodes?.find(
+      (node: any) => node?.field?.name === fieldName
+    );
+    return (
+      fieldValueNode?.text || fieldValueNode?.date || fieldValueNode?.name || ""
+    );
+  };
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.content}>
+        <Input
+          placeholder="Search"
+          value={searchQuery}
+          onChange={(event) =>
+            setSearchQuery((event.target as HTMLInputElement).value)
+          }
+          className={styles.bar}
+          startAdornment={<SearchIcon />}
+        />
+
+        {roadmapData !== null && roadmapData.length > 0 ? (
+          <CardView
+            sortedRoadmapData={sortedRoadmapData}
+            searchQuery={searchQuery}
+          />
+        ) : (
+          <Banner status="info">
+            <BannerContent>No data available</BannerContent>
+          </Banner>
+        )}
+      </div>
+    </div>
+  );
+};
+
+export const CardView = ({ sortedRoadmapData, searchQuery }: CardViewProps) => {
```

Comment:
does this component care that the data is sorted, it does not know, so we could call this 'data' and simplify the naming